libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -124,11 +124,38 @@ AP_Vehicle& vehicle = *AP_Vehicle::get_singleton();
 extern AP_Vehicle& vehicle;
 #endif
 
+#if defined(HAL_GPIO_PIN_WDO)
+static int cnt = 0;
+static bool _sw = false;
+void AP_Vehicle::timer_update(void)
+{
```

Comment:
what is this for? flashing a LED? watchdog?
if this is for watchdog it should be in Scheduler::watchdog_pat()

Reply:
JFB100 has an external watchdog IC.
I would like to leave it as it is, and function name is not clear so changed the name of it.

Reply:
This really should be right next to where we poke the stm32's independent watchdog.  You do not want it on the timer thread - if the main thread stops we really need to trigger watchdogs.

Reply:
This function outputs pulses to the outside at regular intervals while the program is running.
To detect when the main thread has stopped, not the timer thread,
Should it be the main thread?
If so where should I implement it?
(Sorry for my poor knowledge of OS)